### PR TITLE
Add tests custom  `modalTypeBuilder`, `modalDecorator`, `pageContentDecorator`, and restructure the test class

### PR DESCRIPTION
**Pull request by TahaTesser**
Related [Prepare repository for tests requirement](https://github.com/woltapp/wolt_modal_sheet/issues/211)

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors.
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] The package compiles with the minimum Flutter version stated in the [pubspec.yaml](https://github.com/woltapp/wolt_modal_sheet/blob/main/pubspec.yaml#L8)

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[Contributor Guide]: https://github.com/woltapp/wolt_modal_sheet/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: test/wolt_modal_sheet_test.dart [hunk 1]
```diff
@@ -3,57 +3,68 @@ import 'package:flutter_test/flutter_test.dart';
 import 'package:wolt_modal_sheet/wolt_modal_sheet.dart';
 
 void main() {
+  Widget buildSheetWithShow({
```
Comment (ulusoyca) [inline test/wolt_modal_sheet_test.dart:6]: Why does this have to be a method? Can't we just create a `WidgetTestApp` and pump it?

File: test/wolt_modal_sheet_test.dart [hunk 2]
```diff
@@ -147,120 +108,172 @@ void main() {
...
+    ));
+
+    await tester.tap(find.text('Open sheet'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(sheetPageMaterialFinder(tester)),
+        const Size(524.0, 600.0));
+  });
+
+  testWidgets('Custom WoltModalSheet.modalDecorator', (tester) async {
```
Comment (ulusoyca) [inline test/wolt_modal_sheet_test.dart:232]: shall we group the decorator tests?

File: test/wolt_modal_sheet_test.dart (hunk 1 above)
Comment (TahaTesser) [reply test/wolt_modal_sheet_test.dart:6]: It simply to return a widget, we don't need context or state if you're "WidgetTestApp" to be a widget class.

File: test/wolt_modal_sheet_test.dart (hunk 2 above)
Comment (TahaTesser) [reply test/wolt_modal_sheet_test.dart:232]: The goal here is to test all parameters of the constructors, eventually and not just decorators. Generally, I wouldn't group just decorators from a constructor but i can do that do here if you prefer

**Review by ulusoyca — APPROVED** [review]

**Pull request merged**